src/Commands/Extend.php

```diff
@@ -0,0 +1,198 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * This file is part of CodeIgniter Shield.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Shield\Commands;
+use CodeIgniter\CLI\CLI;
+use CodeIgniter\Shield\Commands\Setup\ContentReplacer;
+
+class Extend extends BaseCommand
+{
+    /**
+     * The Command's Name
+     *
+     * @var string
+     */
+    protected $name = 'shield:extend';
+
+    /**
+     * The Command's Description
+     *
+     * @var string
+     */
+    protected $description = 'Extending the Controllers.';
+
+    /**
+     * The Command's Usage
+     *
+     * @var string
+     */
+    protected $usage = 'shield:extend';
+
+    /**
+     * The Command's Arguments
+     *
+     * @var array
+     */
+    protected $arguments = [];
+
+    /**
+     * The Command's Options
+     *
+     * @var array
+     */
+    protected $options = [
+        '-i' => 'The index of shield controllers to be extending in your app.',
+        '-f' => 'Force overwrite ALL existing files in destination.',
+    ];
+
+    protected $sourcePath;
+
+    protected $distPath = APPPATH;
+    private ContentReplacer $replacer;
+
+    private const INFO_MESSAGE    = "  After extending, don't forget to change the route. See https://shield.codeigniter.com/customization/route_config";
+
+    /**
+     * Actually execute a command.
+     *
+     * @param array $params
+     */
+    public function run(array $params)
+    {
+        $this->replacer = new ContentReplacer();
+        $this->sourcePath = __DIR__ . '/../../examples/';
```

Comment:
Why do you need to have examples?
I think we should avoid maintaining both the original controller and the controller in Examples.

Reply:
I created an example to provide a simple way to create an extended controller because in some cases we need to customize the controller. I did not copy all the methods from the source, only some parts that most likely need to be modified.

This is quite helpful for novice users like me, if you guys do not agree with this way, feel free to close this PRs.